### PR TITLE
add null check to db-check and other updates

### PR DESCRIPTION
**Pull request by ryanpmortensen**

### CODE REVIEW

File: apps/workers/db-check/worker.go [hunk 1]
```diff
@@ -36,114 +36,105 @@ func (o *options) newWorker(info string) task.Worker {
...
 	if err != nil {
-		return rows, fmt.Errorf("error getting record count from table %w", err)
+		issues++
+		// there was an error when checking table data add a slack message block with the error
+		m.AddElements(fmt.Sprintf(":octagonal_sign:  *%s* - %s", w.Table, err.Error()))
+		log.Printf("%s - %s\n", w.Table, err.Error())
+	} else {
+		// there are no records, or nothing to process in the table
+		// send a slack message alerting for the table
+		if c == 0 {
```
Comment (jbsmith7741) [inline apps/workers/db-check/worker.go:71]: Combine to flatten logic 
``` go 
else if c == 0 { 

```

File: apps/workers/db-check/worker.go [hunk 2]
```diff
@@ -36,114 +36,105 @@ func (o *options) newWorker(info string) task.Worker {
...
-			fmt.Sprintf("column `%s` value `%s`", w.DtColumn, w.Date.Format("2006-01-02")),
+	c, err := w.GetNullCount(ctx)
+	if err != nil {
+		issues++
+		// there was an error when checking table data add a slack message block with the error
+		m.AddElements(fmt.Sprintf(":octagonal_sign:  *%s; %s* - %s", w.Table, w.Field, err.Error()))
+		log.Printf("%s; %s - %s\n", w.Table, w.Field, err.Error())
+	} else {
+		// null value found - send a slack message alerting for the table & field
+		if c != 0 {
```
Comment (jbsmith7741) [inline apps/workers/db-check/worker.go:113]: combine into single statement `else if`

File: apps/workers/db-check/worker.go [hunk 3]
```diff
@@ -36,114 +36,105 @@ func (o *options) newWorker(info string) task.Worker {
...
-	q := "select count(0) r from " + w.Table + " where date(" + field + ") = $1"
-	err = c.DB.GetContext(ctx, &rows, q, date)
+}
+
+func (w *worker) CheckMissing(ctx context.Context) (task.Result, string) {
+	d := w.Date.Format("2006-01-02")
+	m := w.slack.NewMessage(":radioactive_sign: *task-tools db-check - Missing Data Check - " + d + "*")
+	issues := 0
+
+	c, err := w.GetRecordCount(ctx)
```
Comment (jbsmith7741) [inline apps/workers/db-check/worker.go:62]: I'd prefer `cnt` or `count`

File: apps/workers/db-check/main.go [hunk 4]
```diff
@@ -12,23 +12,22 @@ import (
...
-tolerance - (default 0.05) percent is the allowed percentage of deviation between offset and compare row counts
-              0.05 (5%) means the difference between the row counts can vary by 5 percent and be accepted
-              a greater difference would send an alert, if compare is not provided this check is not done
-
-Example task 
-(table.schema checked for records where my_date = 2022-01-02 and 2022-01-01 has 10% as many records as 2022-01-02):
-  {"type":"task.dbcheck","info":"?db=mydbname&table=table.schema&dt_column=my_date&date=2022-01-02&offset=24h&tolerance=0.1"}`
+db_srv     - database server/source connection
+table      - schema.table name to validate
+type       - type of check
```
Comment (jbsmith7741) [inline apps/workers/db-check/main.go:22]: Would you add a list of the support types here and a description of how they work

File: apps/workers/db-check/main.go [hunk 5]
```diff
@@ -12,23 +12,22 @@ import (
...
-date      - the date to compare data against tolerance level
-offset    - if given, this is the offset duration back from date to compare tolerance
-tolerance - (default 0.05) percent is the allowed percentage of deviation between offset and compare row counts
-              0.05 (5%) means the difference between the row counts can vary by 5 percent and be accepted
-              a greater difference would send an alert, if compare is not provided this check is not done
-
-Example task 
-(table.schema checked for records where my_date = 2022-01-02 and 2022-01-01 has 10% as many records as 2022-01-02):
-  {"type":"task.dbcheck","info":"?db=mydbname&table=table.schema&dt_column=my_date&date=2022-01-02&offset=24h&tolerance=0.1"}`
+db_srv     - database server/source connection
```
Comment (jbsmith7741) [inline apps/workers/db-check/main.go:20]: Is db_src better than db_srv

**Review by jbsmith7741 — COMMENTED** [review]
Can we add some unit tests around the type checks. We could use sqlmock or https://github.com/jbsmith7741/go-tools/tree/main/sqlh to mock around the db calls.

File: apps/workers/db-check/worker.go (hunk 1 above)
Comment (ryanpmortensen) [reply apps/workers/db-check/worker.go]: done

**Review by zJeremiah — APPROVED** [review]

**Pull request merged**